src/entity/posts/utils.ts

```diff
@@ -389,3 +391,75 @@ export const updateUsedImagesInContent = async (
     },
   );
 };
+
+export const addRelatedPosts = async ({
+  entityManager,
+  postId,
+  entries,
+  relationType,
+}: {
+  entityManager: EntityManager;
+  postId: Post['id'];
+  entries: string[];
+  relationType: PostRelationType;
+}): Promise<Post[]> => {
+  if (!entries.length) {
+    return [];
+  }
+
+  const posts = await entityManager.getRepository(Post).findBy({
+    yggdrasilId: In(entries),
+  });
+
+  await entityManager
+    .getRepository(PostRelation)
+    .createQueryBuilder()
+    .insert()
+    .values(
+      posts.map((post) => ({
+        postId,
+        relatedPostId: post.id,
+        relationType,
+      })),
+    )
+    .orIgnore()
+    .execute();
+
+  return posts;
+};
+
+export const relatePosts = async ({
+  entityManager,
+  postId,
+  entries,
+  relationType,
+}: {
+  entityManager: EntityManager;
+  postId: Post['id'];
+  entries: string[];
+  relationType: PostRelationType;
+}): Promise<CollectionPost[]> => {
+  if (!entries.length) {
+    return [];
+  }
+
+  const posts = await entityManager.getRepository(CollectionPost).findBy({
+    yggdrasilId: In(entries),
```

Comment:
Should we search based on this?
Cant we use our internal relations? (Postid)

Feels a bit odd to rely on external systems data

Reply:
It is how it comes from platform, I asked in their DR if it is possible and basically issue is they can not get to post ids.